WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/config/AnalogComplicationConfigRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,754 @@
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.graphics.Color
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffColorFilter
+import android.graphics.drawable.Drawable
+import android.support.wearable.complications.ComplicationHelperActivity
+import android.support.wearable.complications.ComplicationProviderInfo
+import android.support.wearable.complications.ProviderInfoRetriever
+import android.support.wearable.complications.ProviderInfoRetriever.OnProviderInfoReceivedCallback
+import android.util.Log
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.Switch
+import android.widget.Toast
+
+import androidx.core.content.edit
+import androidx.recyclerview.widget.RecyclerView
+
+import com.example.android.wearable.watchfacekotlin.R
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.BackgroundComplicationConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.ColorConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.ConfigItemType
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.MoreOptionsConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.PreviewAndComplicationsConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.UnreadNotificationConfigItem
+import com.example.android.wearable.watchfacekotlin.watchface.AnalogComplicationWatchFaceService
+
+import java.util.concurrent.Executors
+
+/**
+ * Displays different layouts for configuring watch face's complications and appearance settings
+ * (highlight color [second arm], background color, unread notifications, etc.).
+ *
+ *
+ * All appearance settings are saved via [SharedPreferences].
+ *
+ *
+ * Layouts provided by this adapter are split into 5 main view types.
+ *
+ *
+ * A watch face preview including complications. Allows user to tap on the complications to
+ * change the complication data and see a live preview of the watch face.
+ *
+ *
+ * Simple arrow to indicate there are more options below the fold.
+ *
+ *
+ * Color configuration options for both highlight (seconds hand) and background color.
+ *
+ *
+ * Toggle for unread notifications.
+ *
+ *
+ * Background image complication configuration for changing background image of watch face.
+ */
+class AnalogComplicationConfigRecyclerViewAdapter(
+    private val context: Context,
+    watchFaceServiceClass: Class<*>,
+    settingsDataSet: ArrayList<ConfigItemType>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```

Comment:
Can you use a `ListAdapter` here rather than the `RecyclerView.Adapter`?
It should simplify the work you need to do, as you don't need to maintain the list.